### PR TITLE
stablility-tester: add insert on duplicate key update test case.

### PR DESCRIPTION
**Pull request by coocood**

### CODE REVIEW

File: stability-tester/suite/on_dup.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+	return "on_dup"
+}
+
+func (c *OnDupCase) Execute(db *sql.DB, concurrentIndex int) error {
+	c.wg.Add(3)
+	go c.runInsert(db)
+	go c.runDelete(db)
+	go c.runReplace(db)
+	c.wg.Wait()
+	if len(c.errs) != 0 {
```
Comment (shenli) [inline stability-tester/suite/on_dup.go]: Should we use `admin check table` to check the data consistency?

File: stability-tester/suite/on_dup.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	"math/rand"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/pingcap/octopus/stability-tester/config"
+)
+
+// OnDupCase tests insert statement with on duplicate update.
```
Comment (shenli) [inline stability-tester/suite/on_dup.go]: Please add a few comment for the main process of this test.

File: stability-tester/suite/on_dup.go (hunk 1 above)
Comment (coocood) [reply stability-tester/suite/on_dup.go]: Run admin check table periodically is expensive.
If there is data consistency, and we run long enough, one of the workers will fail and stop.

**Comment by coocood** [comment]
@shenli PTAL

**Comment by cwen0** [comment]
It is recommended to write your case according to the new case specification. you can refer to this [pr](https://github.com/pingcap/octopus/pull/52)

**Comment by coocood** [comment]
@cwen0 PTAL

File: stability-tester/config/suite.go [hunk 3]
```diff
@@ -28,6 +28,7 @@ type SuiteConfig struct {
 	CRUD        CRUDCaseConfig        `toml:"crud"`
```
Comment (cwen0) [inline stability-tester/config/suite.go]: you can remove this file, it is useless

File: stability-tester/on_dup/main.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+var defaultPushMetricsInterval = 15 * time.Second
+
+var (
+	logFile    = flag.String("log-file", "", "log file")
+	logLevel   = flag.String("L", "info", "log level: info, debug, warn, error, faltal")
+	user       = flag.String("user", "root", "db username")
+	password   = flag.String("password", "", "user password")
+	dbName     = flag.String("db", "test", "database name")
+	lbService  = flag.String("lb-service", "", "lb")
```
Comment (cwen0) [inline stability-tester/on_dup/main.go]: `pds`, `tidbs`,`tikvs`,`lbService` and `metricAddr` is necessary， I will wrap those extra stuffs into the `case client` in the new version for case to use, but for the current version, these must be added even though you did not use these flags. 😅

**Review by cwen0 — COMMENTED** [review]
rest, LGTM

**Comment by coocood** [comment]
@cwen0 PTAL

**Comment by cwen0** [comment]
LGTM

**Comment by coocood** [comment]
@shenli PTAL

**Comment by shenli** [comment]
LGTM

**Review by shenli — APPROVED** [review]

**Pull request merged**